packages/fxa-payments-server/server/lib/amplitude.test.js

```diff
@@ -0,0 +1,108 @@
+const amplitude = require('./amplitude');
+const log = require('./logging/log')();
+jest.spyOn(log, 'info').mockImplementation(() => {});
+
+jest.mock('../config', () => ({
+  ...jest.requireActual('../config'),
+  get: key => {
+    switch (key) {
+      case 'amplitude':
+        return { enabled: true };
+    }
+  },
+}));
+
+const mocks = {
+  event: {
+    offset: 150,
+    type: 'amplitude.subPaySetup.view',
+  },
+  invalidEventType: {
+    offset: 150,
+    type: 'foo.bar.baz',
+  },
+  data: {
+    deviceId: '0123456789abcdef0123456789abcdef',
+    flowBeginTime: 1570000000000,
+    flowId: '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
+    flushTime: 9002,
+    perfStartTime: 9001,
+    requestReceivedTime: 11000,
+    view: 'product',
+  },
+  request: {
+    headers: {
+      'user-agent':
+        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:72.0) Gecko/20100101 Firefox/72.0',
+    },
+  },
+  requestReceivedTime: 1570000001000,
+};
+
+const expectedOutput = {
```

Comment:
There's no `time` key added to the `expectedOutput` - Is this working as expected?

Reply:
Good catch.  I think jest's `toMatchObject` is not particularly strict: the test is passing because the given object has all the properties of the expected object.  I'll see if there's a better assertion method we can use.

Reply:
@jbuck I think you are correct that this patch didn't fix missing `time` property bug. ☹️

Reply:
I've reopened the issue: https://github.com/mozilla/fxa/issues/3050#issuecomment-547120175.  Thanks, @jbuck!